### DISCUSSION TITLE
Failure to run all DAGs for Airflow in Kubernetes

### DISCUSSION THREAD

**Discussion by xiankgx** [apache/airflow#32057] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Hi, we are running Airflow on Kubernetes on AWS EKS.
It has been running for several years issues. 

All of a sudden it seems none of the DAG runs work and trying to get the logs results in this.

Please help, there are much DAG runs scheduled daily and this error is catastrophic.

*** Falling back to local log
*** Trying to get logs (last 100 lines) from worker pod  ***

*** Unable to fetch logs from worker pod  ***
(400)
Reason: Bad Request
HTTP response headers: HTTPHeaderDict({'Audit-Id': '3bda0e6f-0ea8-4c2b-a950-20a21ed162f0', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'Date': 'Wed, 21 Jun 2023 01:35:12 GMT', 'Content-Length': '136'})
HTTP response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"name must be provided","reason":"BadRequest","code":400}\n'

### What you think should happen instead

Not sure what happened.

### How to reproduce

Unsure

### Operating System

Kubernetes

### Versions of Apache Airflow Providers

Unsure

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Kubernetes

### Anything else

No right now

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by xiankgx** [comment]
![image](https://github.com/apache/airflow/assets/4113258/013e6535-7793-41d5-b617-d0121579310f)

Trying to restart the scheduler and it just seems to keep restarting. What can I do?

**Comment by arjunanan6** [comment]
It looks like your ClusterRole might be broken. I'll take a look at our config and get back to you.

**Comment by hussein-awala** [comment]
Are you using the official helm chart or bitnami chart?
Which executor are you using? is it KubernetesExecutor?
Are you using remote logging? If not, do you have a shared persistent volume between the different services?

I convert it to discussion, and we can reopen it if needed.

**Reply by xiankgx** [reply]
I think we used the official helm chart, but the guys who deployed it are long gone.

**Reply by potiuk** [reply]
Not much the community can do about :( . But maybe a good learning for the future for processes and documetnation for your operations?

Also generally monitoring all the pods/apps and setting the alerting for any apps/pods you run is a good idea :)

**Comment by xiankgx** [comment]
Hi, finally managed to find out the problem.

All DAGs just fail to run, and all the DAG run logs do not help to locate the problem.
But after accessing kubernetes and getting pods, we found that the airflow scheduler kept restarting.
Then, we tried to kill the pod, and it kept restarting as well.
Found out that it is the dags-git-sync container was having an issue. Then, we correlated the problem to BitBucket. The dates and times (when things stopped working and BitBucket said they will rotate the keys) were matching. There was recently a data breach on BitBucket and they required us to change the SSH host key. Ref: https://bitbucket.org/blog/ssh-host-key-changes

So, we then modify the Git secret on kubernetes, and after that all went well again.

Thank you so much everyone for the help.
